File: flutter_gym_app/lib/utils/training/exercise.dart
import 'dart:convert';

import 'package:flutter/cupertino.dart';

class Exercise {
  Exercise({
    @required this.name,
    @required this.reps,
    @required this.sets,
    this.secondForRep,
    this.weight,
    this.urlVideo,
    this.gifOrImage
  });

  String name;
  int reps;
  int sets;
  int secondForRep;
  int weight;
  String urlVideo;
  String gifOrImage;

  Exercise.fromJson(Map<String, dynamic> json)
      : name = json['name'],
        reps = json['reps'],
        sets = json['sets'],
        secondForRep = json['secondForRep'],
        weight = json['weight'],
        urlVideo = json['urlVideo'],
        gifOrImage = json['gifOrImage'];

  Map<String, dynamic> toJson() =>
      {
        'name': name,
        'reps': reps,
        'sets': sets,
        'secondForRep': secondForRep,
        'weight': weight,
        'urlVideo': urlVideo,
        'gifOrImage': gifOrImage
      };
}

File: gymApp_backend/src/main/java/com/buba/gymApp/backend/model/training/Exercise.java
package com.buba.gymApp.backend.model.training;

import java.util.List;
import java.util.UUID;

public class Exercise {
    private final UUID id;
    private final String name;
    private final int reps;
    private final int sets;
    private final int secondForRep;
    private final List<Integer> weights;
    private final String urlVideo;
    private final String gifOrImage;

    public Exercise(UUID id, String name, int reps, int sets, int secondForRep, List<Integer> weights, String urlVideo, String gifOrImage) {
        this.id = id;
        this.name = name;
        this.reps = reps;
        this.sets = sets;
        this.secondForRep = secondForRep;
        this.weights = weights;
        this.urlVideo = urlVideo;
        this.gifOrImage = gifOrImage;
    }

    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public int getReps() {
        return reps;
    }

    public int getSets() {
        return sets;
    }

    public int getSecondForRep() {
        return secondForRep;
    }

    public List<Integer> getWeights() {
        return weights;
    }

    public String getUrlVideo() {
        return urlVideo;
    }

    public String getGifOrImage() {
        return gifOrImage;
    }
}
